### DISCUSSION TITLE
About c++ code structure

### DISCUSSION THREAD

**Discussion by Xiaoyuew** [curv3d/curv#129] [Q&A]
Hi Doug,
Can I ask what is the difference between the 2 folders curv/curv and curv/libcurv?
I am looking through how is the curv program transferred to STL output or rendered to the viewer.
I see that to export STL, the export_mesh.cc called shape.cc in the curv/libcurv, how about rendering? Does rendering also takes a line like "Shape_Program shape(prog)"? I can't find it.

I guess if you add an introduction to the folders of your code, it would be very helpful for more contributors :D
Sincerely
Serena

**Comment by Xiaoyuew** [comment]
Ahhh, I see it, the introduction is in the https://github.com/curv3d/curv/blob/16ba9874b8d0765e189a63687f77172094a72c29/ideas/theory/Implementation. Thanks!

**Reply by doug-moen** [reply]
To make the `Implementation` doc easier to find, I just moved it into the `docs` directory.

**Reply by Xiaoyuew** [reply]
Thank you!
